Mathlib/Topology/Instances/EReal.lean

```diff
@@ -171,6 +170,154 @@ theorem tendsto_nhds_bot_iff_real {α : Type*} {m : α → EReal} {f : Filter α
   nhds_bot_basis.tendsto_right_iff.trans <| by simp only [true_implies, mem_Iio]
 #align ereal.tendsto_nhds_bot_iff_real EReal.tendsto_nhds_bot_iff_real
 
+/-! ### Liminfs and Limsups -/
+
+theorem liminf_le_liminf {α : Type*} {f : Filter α} {u v : α → EReal} (h : u ≤ᶠ[f] v) :
+    liminf u f ≤ liminf v f := Filter.liminf_le_liminf h
+
+theorem limsup_le_limsup {α : Type*} {f : Filter α} {u v : α → EReal} (h : u ≤ᶠ[f] v) :
+    limsup u f ≤ limsup v f := Filter.limsup_le_limsup h
+
+theorem limsup_add_le_lt₂ {α : Type*} {f : Filter α} {u v : α → EReal} {a b : EReal}
+    (ha : limsup u f < a) (hb : limsup v f < b) : limsup (u + v) f ≤ a + b := by
```

Comment:
You should be able to get this?
```suggestion
    (ha : limsup u f < a) (hb : limsup v f < b) : limsup (u + v) f < a + b := by
```

Reply:
Yes, thanks.

Reply:
Is your statement true if you weaken `hb` to a large inequality?

Reply:
No, weakening either `hb` or `ha` results in the statement being false.

With the weakened hypothesis `hb`, only the following statement can be proved:

```lean
limsup (u + v) f ≤ a + b
```

Reply:
Well, that's the current conclusion of the lemma, right? My question is whether, weakening hb, you can still have the conclusion `limsup (u + v) f ≤ a + b`.

Reply:
Done. Thanks.

Reply:
Unless I missed something, you haven't answered this one.

Reply:
Sorry, I have just realised I hadn't submitted the comments (were pending as review comments instead of standard ones).

We have just finished answering this one as follows: 
- [x] Add `add_le_add_right`, `add_le_add_left` and `add_le_add` in Data/Real/EReal.lean
- [x] Use `add_le_add_right ` to prove `limsup_add_le_of_le`